### PR TITLE
Unbonding UI

### PR DESCRIPTION
**Pull request by MaxMillington**
## What is the purpose of the change

UI and logic for unbonding UI

### ClickUp Task

[ClickUp Task URL](https://app.clickup.com/t/8678hgn0w)

## Brief Changelog

Adds UI and logic for unbonding validators 

## Testing and Verifying

This change has been tested locally by rebuilding the website and verified content and links are expected

Screenshot 2023-08-31 at 3 50 29 PM

### CODE REVIEW

File: packages/web/components/cards/stake-info-card.tsx [hunk 1]
```diff
@@ -118,7 +118,7 @@ export const StakeInfoCard: FunctionComponent<{
               outAmountValue ? "opacity-100" : "opacity-0"
             )}
           >
-            {`≈ $ ${outAmountValue || "0"}`}
+            {`≈ ${outAmountValue || "0"}`}
```
Comment (MaxMillington) [inline packages/web/components/cards/stake-info-card.tsx:121]: noticed this UI bug

File: packages/web/pages/stake/index.tsx [hunk 2]
```diff
@@ -268,6 +274,41 @@ export const Staking: React.FC = observer(() => {
 
   const showStakeLearnMore = !isWalletConnected || isNewUser;
 
+  const { unbondingBalances } = unbondingDelegationsQuery;
+  const unbondingInProcess = unbondingBalances.length > 0;
+
+  function groupByCompletionTime(
```
Comment (MaxMillington) [inline packages/web/pages/stake/index.tsx:280]: As mentioned in my slack message, I think this is the only way to get a single stake object

File: packages/web/components/stake/unbonding-in-progress.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+export const UnbondingInProgress: React.FC<{
+  unbondings: {
+    completionTime: string;
+    balance: CoinPretty;
+  }[];
+}> = ({ unbondings }) => {
+  const t = useTranslation();
+  const { priceStore } = useStore();
+
+  function formatUnbondings(
```
Comment (jonator) [inline packages/web/components/stake/unbonding-in-progress.tsx:16]: Is this function definition within the render body needed? It's somewhat hard to follow the logic, given the invocation in the render body below.

File: packages/web/pages/stake/index.tsx [hunk 4]
```diff
@@ -268,6 +274,41 @@ export const Staking: React.FC = observer(() => {
...
+    array: Array<{
+      validatorAddress: string;
+      entries: { completionTime: string; balance: CoinPretty }[];
+    }>
+  ): { completionTime: string; balance: CoinPretty }[] {
+    const groupedObjects: Record<string, CoinPretty> = {};
```
Comment (jonator) [inline packages/web/pages/stake/index.tsx:286]: Personally, I'd prefer a `Map` over using an object, with `Object.*` utils.

Also, perhaps it's possible to use flatMap or reduce over this "array" object? Instead of nested for loops. From experience, the only use case for `for` loops is when you want to save some iterations via `continue`. Otherwise functional programming is preferred.

Also, what is the content of completion time. Is it an ISO string? IMO proper Date or dayjs duration objects offer more type safety then string representations. These object ideally should be created at the query store later and consumed by the view.

Comment (MaxMillington) [reply packages/web/pages/stake/index.tsx:286]: I can get rid of the nested loop. Good catch. 

Map and Object aren't going to have any functional difference here, since I am just storing strings and values.

`completionTime` is just whatever string is returned by the chain. I don't want to change the type since all I'm doing is a groupBy.

File: packages/web/components/stake/unbonding-in-progress.tsx (hunk 3 above)
Comment (MaxMillington) [reply packages/web/components/stake/unbonding-in-progress.tsx:16]: given that this is just formatting for this view, it didn't seem like it was worth it to extract this function. Since this view is only rendered once, any performance difference is negligible.

**Review by jonator — APPROVED** [review]
